sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -108,7 +108,7 @@ private AppendBlobOutputStream(final AppendBlobAsyncClient client,
 
             return client.appendBlockWithResponse(blockData, writeLength, appendBlobAccessConditions)
                 .then()
-                .onErrorResume(t -> t instanceof IOException || t instanceof StorageException, e -> {
+                .onErrorResume(t -> t instanceof IOException || t instanceof BlobStorageException, e -> {
                     this.lastError = new IOException(e);
                     return null;
```

Comment:
Does this work? I noticed all these `null` returns. iirc, it'll throw an exception if a null is returned.

Reply:
I'll change these to return Mono.empty